drivers/dac/dac_samd5x.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2024 Daikin Comfort Technologies North America, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT atmel_samd5x_dac
+
+#include <errno.h>
+
+#include <zephyr/drivers/dac.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dac_samd5x, CONFIG_DAC_LOG_LEVEL);
+
+#define DAC_CHANNEL_NO  2
+#define DAC_RESOLUTION 12
+
+#define SAMD5X_DAC_REFSEL_0 DAC_CTRLB_REFSEL_VREFPU
+#define SAMD5X_DAC_REFSEL_1 DAC_CTRLB_REFSEL_VDDANA
+#define SAMD5X_DAC_REFSEL_2 DAC_CTRLB_REFSEL_VREFPB
+#define SAMD5X_DAC_REFSEL_3 DAC_CTRLB_REFSEL_INTREF
+
+struct dac_samd5x_channel_cfg{
+	uint8_t oversampling;     /* Oversampling ratio */
+	bool dither_mode;         /* Dithering */
+	uint8_t refresh_period;   /* Refresh period */
+	bool run_in_standby;      /* Run in standby mode */
+	bool filter;           	  /* External Filter mode */
+	uint8_t current_control;  /* Current control */
+	bool left_adjusted;       /* Left-adjusted data */
+	bool enable_channel;	  /* Enable channel */
+};
+
+struct dac_samd5x_cfg{
+	Dac *regs;
+	const struct pinctrl_dev_config *pcfg;
+	volatile uint32_t *mclk;
+	uint32_t mclk_mask;
+	uint32_t gclk_gen;
+	uint16_t gclk_id;
+	uint8_t refsel;
+	bool diff_mode;
+	struct dac_samd5x_channel_cfg channel_cfg[DAC_CHANNEL_NO];
+};
+
+/* Write to the DAC. */
+static int dac_samd5x_write_value(const struct device *dev, uint8_t channel,
+				uint32_t value)
+{
+	const struct dac_samd5x_cfg *const cfg = dev->config;
+	Dac *regs = cfg->regs;
+
+	if (value >= BIT(12)) {
+		LOG_ERR("value %d out of range", value);
+		return -EINVAL;
+	}
+	if (channel >= DAC_CHANNEL_NO) {
+		return -EINVAL;
+	}
+	
+	regs->DATA[channel].reg = (uint16_t)value;
+	
+	if (channel == 0){
+		while ((regs->SYNCBUSY.reg & DAC_SYNCBUSY_DATA0) == DAC_SYNCBUSY_DATA0) {
+			/* Wait for synchronization */
+		}
+	} else {
+		while ((regs->SYNCBUSY.reg & DAC_SYNCBUSY_DATA1) == DAC_SYNCBUSY_DATA1) {
+			/* Wait for synchronization */
+		}
+	}
+	
+	return 0;
+}
+
+/*
+* Does nothing, DAC channels properties are set up in dac_samd5x_init. 
+*/
+static int dac_samd5x_channel_setup(const struct device *dev,
+				const struct dac_channel_cfg *channel_cfg)
+{
+	/* Does nothing */
```

Comment:
This function should at least check if the desired resolution for the given channel is valid.